synapse/lib/ast.py

```diff
@@ -1109,6 +1140,14 @@ async def run(self, runt, genr):
 
 class WhileLoop(Oper):
 
+    def hasAstClass(self, clss):
+        if (hasast := self.hasast.get(clss)) is not None:
+            return hasast
+
```

Comment:
Small nit, but perhaps we could implement a `_hasAstClass()` which implements the get without checking the cache and have the `hasAstClass()` API check the cache, invoke the `_has...` method and store in `self.hasast` to minimize dup code?

Reply:
split hasAstClass and cleaned it up